src/hww.c

```diff
@@ -163,11 +189,50 @@ static void _msg(const Packet* in_packet, Packet* out_packet, const size_t max_o
     }
 
     // Process protofbuf/noise api calls.
-    if (!bb_noise_process_msg(in_packet, out_packet, max_out_len, commander)) {
+    if (!bb_noise_process_msg(in_req, out_rsp, commander)) {
         workflow_status_create("Could not\npair with app", false);
     }
 }
 
+static void _msg(const Packet* in_packet, Packet* out_packet, const size_t max_out_len)
+{
+    if (in_packet->len == 0) {
+        out_packet->data_addr[0] = HWW_RSP_NACK;
+        out_packet->len = 1;
+        return;
+    }
+    hww_packet_req_t decoded = {.cmd = in_packet->data_addr[0],
+                                .buffer = {
+                                    .data = in_packet->data_addr + 1,
+                                    .len = in_packet->len - 1,
+                                }};
+    hww_packet_rsp_t response = {
+        .status = HWW_RSP_NACK,
+        .buffer = {.data = out_packet->data_addr + 1, .len = 0, .max_len = max_out_len - 1}};
+    switch (decoded.cmd) {
+    case HWW_REQ_NEW:
+        _process_packet(&decoded.buffer, &response.buffer);
+        response.status = HWW_RSP_ACK;
+        break;
+    case HWW_REQ_CANCEL:
+        /* We don't have anything to cancel yet. */
+        response.status = HWW_RSP_NACK;
+        break;
+    case HWW_REQ_RETRY:
+        /* We don't support async retries yet. */
+        response.status = HWW_RSP_NACK;
+        break;
+    default:
+        break;
+    }
+    out_packet->data_addr[0] = response.status;
+    if (response.status == HWW_RSP_ACK) {
+        out_packet->len = response.buffer.len + 1;
```

Comment:
you can use this line regardless of the response status, right? response.buffer.len is 0 if nothing was written to it.